File: src/main/java/github/macro/pieces/Scholar.java
package github.macro.pieces;

import github.macro.Direction;
import github.macro.console.Colour;
import org.jetbrains.annotations.NotNull;

import static github.macro.Direction.*;

/**
 * Created by Macro303 on 2018-02-09.
 */
public class Scholar extends Piece {
	@NotNull
//	private static final String symbol = "\uD83D\uDCDA";
	private static final String symbol = "C";
	private static final int maxDistance = 2;
	@NotNull
	private static final Direction[] validDirections = new Direction[]{NORTH, NORTH_EAST, EAST, SOUTH_EAST, SOUTH, SOUTH_WEST, WEST, NORTH_WEST};

	public Scholar(@NotNull Colour colour) {
		super(colour, maxDistance, symbol, validDirections);
	}

	public Scholar() {
		super(maxDistance, symbol, validDirections);
	}

	@Override
	public String toString() {
		return "Scholar{} " + super.toString();
	}
}

File: src/main/kotlin/github/macro/pieces/Scholar.kt
package github.macro.pieces

import github.macro.Direction
import github.macro.players.Player

/**
 * Created by Macro303 on 2018-02-09.
 */
class Scholar(player: Player) : Piece(
	player = player,
	distance = 2,
	symbol = "C",
	directions = arrayOf(
		Direction.NORTH,
		Direction.NORTH_EAST,
		Direction.EAST,
		Direction.SOUTH_EAST,
		Direction.SOUTH,
		Direction.SOUTH_WEST,
		Direction.WEST,
		Direction.NORTH_WEST
	)
) {
//	private static final String symbol = "\uD83D\uDCDA";
//	private static final String symbol = "📚";
//	private static final String symbol = "C";

	override fun toString(): String = "Scholar() ${super.toString()}"
}